backup_github.py

```diff
@@ -31,114 +40,148 @@ def __enter__(self):
 
     def __exit__(self, type, value, tb):
         os.chdir(self.original_dir)
-        
+
+
 def credentials():
-    config_file = os.path.join(os.environ.get("HOME"), ".githubbackup_creds.yaml")
+    config_file = os.path.join(os.environ.get("HOME"),
+                               ".githubbackup_creds.yaml")
     if not os.path.exists(config_file):
         config_file = os.path.join(os.environ.get("GITHUBBACKUP_CREDS"))
     with open(config_file) as f:
-        conf = yaml.load(f)
+        conf = yaml.load(f, Loader=yaml.SafeLoader)
     return conf
 
-def backup(user, password, dest):
-    """Performs a backup of all the public repos in user's GitHub account on dest
+
+def backup(user, password, organizations, dest):
+    """Performs a backup of all the accessible repos in given organizations
     """
-    if not password is None:
-        gh = Github(login=user, user=user, password=password)
-        repos = gh.repos.list(type='all')
-    if password is None or repos.all() == []:
-	print "No valid github credentials provided. Private repos will not be copied!"
-        logging.info("No valid github credentials provided. Private repos will not be copied!")
-        gh = Github()
-        repos = gh.repos.list(type='all', user=user)
-
-    for repo in repos.all():
+    if password is None or user is None:
+        logger.error("No valid github credentials provided. Exiting!")
+        sys.exit(-1)
+    if password is not None:
+        gh = Github(user, password)
+        repos_l = []  # list of repos iterators
+        for org in organizations:
+            gh_org = gh.get_organization(org)
```

Comment:
Just to be picky, maybe consider renaming some variables here? 😉 Perhaps github_instance instead of gh, repositories instead of repos_l, organization instead of org, and github_organization instead of gh_org, or something similar?

Reply:
Yes, I might have some personal abbreviations which I don't realise are not common to everyone